pilot/pkg/proxy/envoy/discovery.go

```diff
@@ -117,9 +117,14 @@ var (
 	// Default value is 1 second, or the value of PILOT_CACHE_SQUASH env
 	clearCacheTime = 1
 
-	// V2ClearCache is a function to be called when the v1 cache is cleared. This is used to
-	// avoid adding a circular dependency from v1 to v2.
-	V2ClearCache func()
+	// Push is used to request a push, when config changes. This is used to
+	// avoid adding a circular dependency from v1 to v2 - the method is implemented
+	// in the ADS server.
+	Push func(fullPush bool, edsUpdates map[string]*model.ServiceShards)
+
+	// BeforePush is called before a push. Like Push, it is an artifact of the split
+	// of discovery from v2/discovery and the migration from v1.
```

Comment:
we don't have v1. So whats the hold up?

Reply:
In master - no problem. In 1.0 - this PR is already far too big.